### PR TITLE
bazel: add docs page for bazel & rust

### PR DESCRIPTION
**Pull request by Strum355**
Add dedicated page for Rust & Bazel, linking to the FAQ in more places, and some minor fixes to links etc

### Test plan

`pnpm run dev`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,3 +36,4 @@ next-env.d.ts
 
 # contentlayer
 .contentlayer
+public/search.json
```
Comment (Strum355) [inline .gitignore:39]: This file keeps getting changed locally, Im not sure if its meant to be checked-in or not but the `git diff` noise is a bit annoying @MaedahBatool

File: docs/dev/background-information/bazel/rust.mdx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+## TL;DR
+
+- Commands:
+  - `CARGO_BAZEL_REPIN=1 bazel sync --only=crate_index` after changing workspace members or dependencies.
+    - Setting `CARGO_BAZEL_ISOLATED=0` can be set if repinning is too slow, if doing it frequently in local development.
+  - `CARGO_BAZEL_REPIN=<crate name> bazel sync --only=crate_index` after adding/updating a dependency.
+
+## Overview
+
+The rules interfacing Rust are named [`rules_rust`](https://github.com/bazelbuild/rules_rust/) and they provide all the plumbing to call the Rust compiler, run the tests
```
Comment (burmudar) [inline docs/dev/background-information/bazel/rust.mdx:14]:
```suggestion
The rules interfacing Rust are named [`rules_rust`](https://github.com/bazelbuild/rules_rust/) and they provide all the plumbing to call the Rust compiler, run the tests.
```

File: docs/dev/background-information/bazel/rust.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Bazel and Rust works slightly differently to Bazel and Go. Unlike with Go, `BUILD.bazel` files are not updated with `bazel configure` (they must be updated/configured/created by hand), and instead of there being one `BUILD.bazel` file per directory, it's one per workspace member.
+There exists [gazelle_rust](https://github.com/Calsign/gazelle_rust), a plugin for [Gazelle](https://github.com/bazelbuild/bazel-gazelle) which is invoked via `bazel configure`, that may address the first point but we have not decided to invest in using it at the time of writing.
```
Comment (burmudar) [inline docs/dev/background-information/bazel/rust.mdx:17]:
```suggestion
There exists [gazelle_rust](https://github.com/Calsign/gazelle_rust), a plugin for [Gazelle](https://github.com/bazelbuild/bazel-gazelle) which is invoked via `bazel configure`, that may address the first point but we have decided not to invest in using it at the time of writing.
```

File: docs/dev/background-information/bazel/rust.mdx (hunk 2 above)
Comment (Strum355) [reply docs/dev/background-information/bazel/rust.mdx:14]: Oh this is an unfinished sentence, gotta finish it lol

Comment (burmudar) [reply docs/dev/background-information/bazel/rust.mdx:14]: oh whoops 🤣 ok I'll refrain from commenting now 😄

File: docs/dev/background-information/bazel/go.mdx [hunk 4]
```diff
@@ -20,10 +22,10 @@ See also: [How-to use `go:generate` with Bazel](/dev/background-information/baze
 
 ## Overview
 
-Bazel and Go is pretty straightforward. With a few minor exceptions, you can still use your normal Go tooling and only care about Bazel
+Bazel and Go is pretty straightforward. With a few minor exceptions, you can still use your normal Go tooling and only care about Bazel 
 in CI, as long as you remember to run `bazel configure` before pushing your changes.
 
-The rules interfacing Go are named `rules_go` and they provide all the plumbing to call the Go compiler and run the tests. Gazelle is the tool that parses you Go code and the `go.mod` file to scaffold the build files for you. It's tempting to think about it as a generator that could be fully automated in CI: it is not the case.
+The rules interfacing Go are named [`rules_go`](https://github.com/bazelbuild/rules_go) and they provide all the plumbing to call the Go compiler and run the tests. [Gazelle](https://github.com/bazelbuild/bazel-gazelle) is the tool that parses you Go code and the `go.mod` file to scaffold the build files for you. It's tempting to think about it as a generator that could be fully automated in CI: it is not the case.
```
Comment (burmudar) [inline docs/dev/background-information/bazel/go.mdx:28]:
```suggestion
The rules interfacing Go are named [`rules_go`](https://github.com/bazelbuild/rules_go) and they provide all the plumbing to call the Go compiler and run the tests. [Gazelle](https://github.com/bazelbuild/bazel-gazelle) is the tool that parses your Go code and the `go.mod` file to scaffold the build files for you. It's tempting to think about it as a generator that could be fully automated in CI: it is not the case.
```

**Review by burmudar — APPROVED** [review]

**Pull request merged**